base-template.sh

```diff
@@ -592,6 +590,48 @@ call_script() {
     return $?
 }
 
+#####################################################
+# Show a prompt with the text `$2` with
+#   hint text `$3` with
+#   options `$4` in form of e.g. `Y/a/n/d` and
+#   optional long options [optional]:
+#   e.g.
+#    `$5-$8` : "Yes" "All" "None" "Disable"
+#   First capital short option character is treated
+#   as default.
+#
+#####################################################
+show_prompt() {
+    DIALOG_TOOL="$(get_tool_script "dialog")"
+    VARIABLE="$1"
+    shift
+    TEXT="$1"
+    shift
+    HINT_TEXT="$1"
+    shift
+    SHORT_OPTIONS="$1"
+    shift
+
+    if [ "$DIALOG_TOOL" != "" ]; then
+        ANSWER=$(call_script "$DIALOG_TOOL" "githook::" "$TEXT" "$SHORT_OPTIONS" "$@")
+        # shellcheck disable=SC2181
+        if [ $? -ne 0 ]; then
+            echo "! Dialog tool did not succeed -> Abort."
+            exit 1
+        fi
+        if ! echo "$SHORT_OPTIONS" | grep -q "$ANSWER"; then
+            echo "! Dialog tool did return wrong answer $ANSWER -> Abort."
+            exit 1
+        fi
+        eval "$VARIABLE"="$ANSWER"
```

Comment:
`eval` makes me a bit nervous but haven't found anything POSIX as an alternative. Could we at least validate it's nothing dangerous, like only uppercase characters with underscores?
I think this should do the trick:
```
if ! echo "$VARIABLE" | grep -qE "^[A-Z_]+\$"; then
    echo "! Invalid variable name: $VARIABLE"
    exit 1
fi
```

Reply:
Jeah, makes me nervous as well, I couldnt find something better as well unfortunately.
Lets do this, jup!